controllers/drplacementcontrol_controller.go

```diff
@@ -1759,3 +1761,30 @@ func addOrUpdateCondition(conditions *[]metav1.Condition, conditionType string,
 
 	return false
 }
+
+// Initial creation of the DRPC status condition. This will also preserve the ordering of conditions in the array
+func ensureDRPCConditionsInited(conditions *[]metav1.Condition, observedGeneration int64, message string) {
+	const DRPCTotalConditions = 2
+	if len(*conditions) == DRPCTotalConditions {
+		return
+	}
+
+	time := metav1.NewTime(time.Now())
+
+	setStatusConditionIfNotFound(conditions, metav1.Condition{
+		Type:               rmn.ConditionAvailable,
+		Reason:             string(rmn.Initiating),
+		ObservedGeneration: observedGeneration,
+		Status:             metav1.ConditionTrue,
+		LastTransitionTime: time,
+		Message:            message,
+	})
+	setStatusConditionIfNotFound(conditions, metav1.Condition{
+		Type:               rmn.ConditionPeerReady,
+		Reason:             string(rmn.Initiating),
+		ObservedGeneration: observedGeneration,
+		Status:             metav1.ConditionTrue,
+		LastTransitionTime: time,
+		Message:            message,
+	})
```

Comment:
This works only if both conditions are missing, only rmn.ConditionAvailable exists, or both exist. Is it possible that only rmn.ConditionPeerReady exits and it is at index 0 when this is called?

Reply:
While true, this does not occur at present, and not post this PR. The added logic to copy the existing condition in the right order maybe unwanted post this change.

Reply:
We can add a comment describing this assumption. Typically you assert() about this so if you assumption is wrong you find it during testing.